dev-docs/source/VSCode.rst

```diff
@@ -0,0 +1,325 @@
+.. _VSCode:
+
+.. |extensions| image:: /images/VSCode/extension-button.png
+.. |debug| image:: /images/VSCode/debug-button.png
+.. |debug-cog| image:: /images/VSCode/debug-cog-button.png
+
+======
+VSCode
+======
+
+.. contents::
+    :local:
+
+Introduction
+============
+This is a guide aimed at the developer who does not want a heavy complicated IDE
+for C++ development on any platform that VSCode supports
+
+Prerequisites
+=============
+- An installed copy of VSCode - `download <https://code.visualstudio.com/>`_.
+- A clone of the main Mantid repo
+- Follow the :ref:`GettingStarted` guide to get a:
+- Built copy of Mantid (In "Debug" mode for any debugging option not "Release")
+- GDB for Linux/OSX debugging (GDB can be switched out for LLDB at any stage on OSX) or MSVC (Visual Studio) for Windows
+
+How to get Started
+==================
+- Run VSCode
+- Click File->Open Folder
+- Navigate to your Mantid source directory and select it
+- Install the required extensions (see below)
+- For code editing you are good to go!
+
+Extensions
+==========
+All Extensions have been tested working on Ubuntu 18.04 and Ubuntu 19.04,
+however most, if not all, extensions should be cross-platform.
+
+Install extensions either by running the commands given on the marketplace or by
+clicking on this Icon |extensions|.
+
+Required
+--------
+- `C/C++ <https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools>`_
+    - Required for C++ linting, debugging etc
+- `Python <https://marketplace.visualstudio.com/items?itemName=ms-python.python>`_
+    - Required to run any python code, debug etc
+
+Recommended
+-----------
+- `Clang-Format <https://marketplace.visualstudio.com/items?itemName=xaver.clang-format>`_
+    - Allows auto formatting using the clang format version installed on your machine
+- `CMake <https://marketplace.visualstudio.com/items?itemName=twxs.cmake>`_
+    - Allows language highlighting and auto fill for CMake
+- `Doxygen Document Generator <https://marketplace.visualstudio.com/items?itemName=cschlosser.doxdocgen>`_
+    - Once a function is written with this extension it can generate parts of the doxygen comments itself
+- `GitLens <https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens>`_
+    - Very powerful source control assistant
+- `Visual Studio Intellicode <https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.vscodeintellicode>`_
+    - This improves your code suggestions in certain languages
+
+Features
+========
+
+Auto Formatting
+---------------
+With a formatter installed as an extension it is possible to format any open file in VSCode.
+It is recommended to use `Clang-Format <https://marketplace.visualstudio.com/items?itemName=xaver.clang-format>`_.
```

Comment:
We require `clang-format`.

Reply:
I understand that, however turning on auto-formatting in VSCode is optional, there are other ways to format it, I also was just recommending one of the Clang-format extensions.

Reply:
That should be more obvious, right now it reads "formatting is a thing, you should probably do it with clang-format".

Reply:
I have made a change to how the test reads.